### PR TITLE
Validate /offer dates

### LINKED ISSUE #45: It is possible to supply invalid date to POST /offer
## Current behavior
POST /offer responds with 200 and inserts the offer, when supplied with `date_from > date_to` or negative date values as `OfferDto` fields. 

## Expected behavior
POST /offer should respond with 400 - bad request, when supplied with invalid date values.

### PR DESCRIPTION
**Pull request by Xertes0**

### CODE REVIEW

File: server/src/dto/OfferDto.hh [hunk 1]
```diff
@@ -19,9 +19,9 @@ class OfferDto : public oatpp::DTO
 
     DTO_FIELD(UInt64, place_id, "place_id");
 
-    DTO_FIELD(Int64, date_from, "date_from");
+    DTO_FIELD(UInt64, date_from, "date_from");
```
Comment (papaj-na-wrotkach) [inline server/src/dto/OfferDto.hh:22]: Why the change? Theoretically the timestamp can be negative...

File: server/src/service/OfferService.cc [hunk 2]
```diff
@@ -8,9 +8,19 @@ OfferService::createShared()
     return std::make_shared<OfferService>();
 }
 
+void
+validateDateHTTP(UInt64 const& date_from, UInt64 const& date_to)
+{
+    OATPP_ASSERT_HTTP(date_from <= date_to,
+                      Status::CODE_400,
+                      "'date_from' cannot occur later than 'date_to'");
```
Comment (papaj-na-wrotkach) [inline server/src/service/OfferService.cc]: should be changed to sth more simple like "Invalid date"

**Review by papaj-na-wrotkach — CHANGES_REQUESTED** [review]
HTTP 400 status code should be documented in swagger

File: server/src/dto/OfferDto.hh (hunk 1 above)
Comment (Xertes0) [reply server/src/dto/OfferDto.hh:22]: > Theoretically the timestamp can be negative...

How can they be negative? Just to be sure, we are using Unix timestamps, right?

File: server/src/service/OfferService.cc (hunk 2 above)
Comment (Xertes0) [reply server/src/service/OfferService.cc]: > should be changed to sth more simple like "Invalid date"

I think, it's better to point out the error as precisely as we can.

File: server/src/dto/OfferDto.hh (hunk 1 above)
Comment (papaj-na-wrotkach) [reply server/src/dto/OfferDto.hh:22]: If the timestamp is negative it means that it refers to date before 00:00:00 UTC on 1 January 1970

File: server/src/service/OfferService.cc (hunk 2 above)
Comment (papaj-na-wrotkach) [reply server/src/service/OfferService.cc]: let's stick to simple messages. It is very different from the others stylistically

Comment (papaj-na-wrotkach) [reply server/src/service/OfferService.cc]: How about "Time range cannot be negative" or "Invalid time range"

File: server/src/dto/OfferDto.hh (hunk 1 above)
Comment (Xertes0) [reply server/src/dto/OfferDto.hh:22]: > If the timestamp is negative it means that it refers to date before 00:00:00 UTC on 1 January 1970

Should we allow users to post offers to the past?

Comment (papaj-na-wrotkach) [reply server/src/dto/OfferDto.hh:22]: Why not `¯\_(ツ)_/¯`

What if a time traveler would like to use this project in multiple time lines - I mean that you could buy an offer which will be posted in a future?
